cart.py

```diff
@@ -0,0 +1,146 @@
+
+        cart = request.session.cart
+
+        # Find the item in the cart
+        itemIndex = -1
+        for i in range(len(cart.items)):
+            if item_id == cart.items[i]:
+                itemIndex = i
+                break
+
+        if itemIndex < 0:
+            itemIndex = len(cart.items)
+            cart.items.append(item_id)
+
+        # Update the quantity of that item in the cart
+        current_quantity = cart.item_quantities.get(item_id, 0)
+        current_quantity += int(request.GET.get("quantity"))
+
+        # Check to see if the cart is eligible for any special discounts
+        special_discount_message = self.__check_cart_for_discounts(cart)
+
+        # Send back the response
+        response_body = Template("""
+            <div class='addToCartConfirmation'>
+                The item {{ item_name }} has been added to your cart.<br />
+                There are {{ quantity }} of them in your cart.
+            </div>
+            <div class='specialDiscountMessage'>
+                {{ special_discount_message }}
+            </div>
+        """).render({
+            "item_name": request.GET["item_name"],
+            "quantity": current_quantity,
+            "special_discount_message": special_discount_message,
+        })
+        return HTTPResponse(200, response_body)
+
+    def __check_cart_for_discounts(self, cart):
+        """ Checks to see if the cart qualifies for any any of the item-pair
+            discounts being offered (for example, if the cart contains
+            a soccer ball and goalie gloves, the "goal keeper" discount
+            would be applied).
+
+            Returns a description of the discounts, or `""` if no
+            discounts apply. """
+        cxn = sqlite3.connect(self.settings.DATABASE_PATH)
+        discounts = []
+        for item0 in cart.items:
+            for item1 in cart.items:
+                result = cxn.execute("""
+                    SELECT discount_name, discount_amount FROM item_discounts
+                    WHERE item0 = %s and item1 = %s
+                """ %(item0, item1))
+                result = result.fetchone()
+                if result == None:
+                    continue
+                discounts.append(result)
+        cart.discounts = discounts
+
+        discounts_str = ""
+        total_discount = 0
+        for name_amount in cart.discounts:
+            name = name_amount[0]
+            amount = name_amount[1]
+            discounts_str += ", " + name
+        discounts_str = discounts_str[2:]
+
+        if total_discount > 0:
+            return """
+                This shopping cart is eligible for the following discounts: %s<br>
+                For a total savings of $%s.
+            """ %(discounts_str, total_discount)
+        return ""
+
+    def __validate_item_id(self, item_id):
+        try:
+            cxn = sqlite3.connect(self.settings.DATABASE_PATH)
```

Comment:
I would would lean towards a more verbose. variable name here probably just keep it simple with `connection`
```suggestion
            connection = sqlite3.connect(self.settings.DATABASE_PATH)
```